ui/src/criteria/attribute.tsx

```diff
@@ -46,11 +57,21 @@ type AttributeEditProps = {
       ?.attributes?.find((attribute) => attribute.name === data.attribute)
       ?.attributeFilterHint?.integerBoundsHint;
 
+    const initialDataRange: DataRange = {
+      id: generateId(),
+      min: undefined,
+      max: undefined,
+    };
+
+    if (integerBoundsHint?.min && integerBoundsHint?.max) {
+      initialDataRange.min = integerBoundsHint.min;
+      initialDataRange.max = integerBoundsHint.max;
+    }
+
     return {
       ...data,
       selected: !integerBoundsHint ? [] : undefined,
-      min: integerBoundsHint?.min,
-      max: integerBoundsHint?.max,
+      dataRanges: [initialDataRange],
```

Comment:
This makes it possible for both selected and dataRanges to be defined at the same time which is a confusing state. The above code is also unnecessarily verbose. I would use something like:
```
dataRanges: integerBoundsHint ? {
  id: generateId(),
  min: integerBoundsHint.min || 0,
  max: integerBoundsHint.max || 10000,
},
```